tests/test_core.py

```diff
@@ -24,181 +24,221 @@
 
 """Test Matcher core."""
 
-from helpers import data, simple_data, single_query
+from __future__ import absolute_import, print_function
 
-from invenio_base.wrappers import lazy_import
+import mock
+import pytest
 
 from invenio_matcher.core import _merge, _parse, execute, get_queries
 from invenio_matcher.errors import InvalidQuery, NotImplementedQuery
 
-from invenio_testing import InvenioTestCase
+from .helpers import empty_search_result
 
-import mock
 
-import pytest
+def test_queries(app, matcher_config):
+    """Get defined queries from configuration."""
+    app.config.update(dict(MATCHER_QUERIES=matcher_config))
+    with app.app_context():
+        expected = [{'type': 'exact', 'match': 'title'}]
+        result = get_queries(index='records', doc_type='record')
 
+        assert expected == result
 
-Record = lazy_import('invenio_records.api.Record')
 
+def test_execute_exact(app, simple_record, mocker):
+    """Dispatch the query of type exact."""
+    from invenio_records import Record
+    mocker.patch('invenio_matcher.engine.search', empty_search_result)
 
-class TestMatcherCore(InvenioTestCase):
+    with app.app_context():
+        query = {'type': 'exact', 'match': 'title'}
+        index = "records"
+        doc_type = "record"
 
-    """Matcher: test core."""
+        record = Record(simple_record)
 
-    def setup_class(self):
-        self.data = data()
-        self.simple_data = simple_data()
+        expected = []
+        result = execute(index, doc_type, query, record)
 
-    @mock.patch('invenio_matcher.core.exact')
-    def test_execute_exact(self, exact):
-        """Dispatch the query of type exact."""
-        query = {'type': 'exact', 'match': 'titles.title'}
-        record = Record(self.data)
+        assert result == expected
 
-        execute(query, record)
 
-        exact.assert_called_with(match='titles.title', values=['foo bar'])
+def test_execute_fuzzy(app, simple_record, mocker):
+    """Dispatch the query of type fuzzy."""
+    from invenio_records import Record
```

Comment:
Nit: imports inside of tests where needed to work around the previous registry madness, now they could be dropped.